docs/specifications/tests/RequirementsAndNormalizationOfDomainNames.md

```diff
@@ -0,0 +1,476 @@
+cannot be used for Zonemaster testing.
+
+If the outcome (see [Outcome(s)](#Outcomes)) is not "fail" then *Domain Name* in
+normalized form is returned to be used as input value for Zonemaster test cases.
+
+See the details in the [Detailed requirements] section below.
+
+
+### References
+
+The following references are consulted for this specification:
+
+* [RFC 1034]
+* [RFC 1035]
+* [RFC 1123]
+* [RFC 2317]
+* [RFC 2782]
+* [RFC 5890]
+* [RFC 5895]
+* [Unicode TR 46]
+
+
+## Scope
+
+This specification only tests and creates a normalized form of the domain name
+(zone name or name server name).
+
+In this specification, ASCII is identical to the first 128 characters in
+[Unicode] (0000..007F).
+
+[RFC 1123][RFC 1123#2.1], section 2.1, specifies that a domain name label
+may not start or end with a HYPHEN-MINUS ("-"), only digit or letter. This
+restriction on HYPHEN-MINUS is disregarded in this specification and is assumed
+to be handled in test case [Syntax02].
+
+The use of the SOLIDUS ("/") and the LOW LINE ("_") in domain name is discussed
+in the section "[ASCII domain name](#ASCII-domain-name)" below. Any restrictions
+on where in the domain name or label those could or should be used are
+disregarded in this specification, and are assumed to be handled in test cases
+[Syntax01] and [Syntax02].
+
+
+## Inputs
+
+* "Domain Name" - The domain name to be tested and normalized according to this
+  specification. It must be a non-empty string of [Unicode] characters.
+
+## Summary
+
+In the specification there are six scenarios that will result in the domain name
+not being usable, i.e. it cannot be used for Zonemaster testing. Each scenario
+is here listed with a message tag, level (always CRITICAL in this specification),
+suitable argument to be used in the same descriptive message and a message that
+can be returned to the user.
+
+Message Tag           | Level    | Arguments | Message ID for message tag
+:---------------------|:---------|:----------|:---------------------------------
+INITIAL_DOT           | CRITICAL |           | Domain name starts with dot.
+REPEATED_DOTS         | CRITICAL |           | Domain name has repeated dots.
+INVALID_ASCII         | CRITICAL | dlabel    | Domain name has an ASCII label ("{dlabel}") with a character not permitted.
+INVALID_U_LABEL       | CRITICAL | dlabel    | Domain name has a non-ASCII label ("{dlabel}") which is not a valid U-label.
+LABEL_TOO_LONG        | CRITICAL | dlabel    | Domain name has a label that is too long (more than 63 characters), "{dlabel}".
+DOMAIN_NAME_TOO_LONG  | CRITICAL |           | Domain name is too long (more than 253 characters with no final dot).
+
+The value in the Level column is the default severity level of the message. Also
+see the [Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
```

Comment:
Looking at the logentry_args.md document, it looks like the `dlabel` is not under the "Defined arguments" section. Maybe update the document as well.

Reply:
Will do.

Reply:
Chose to define "label" (not "dlabel") in PR https://github.com/zonemaster/zonemaster-engine/pull/1126.

By commit 78732ed to this PR, "dlabel" is replaced by "label".